src/components/gltf-model-plus.js

```diff
@@ -233,6 +244,17 @@ async function inflateComponents(inflatedEntity, indexToEntityMap) {
       }
     }
 
+    const materialComponents = getHubsMaterialComponents(object3D);
+
+    if (materialComponents && el) {
+      for (const prop in materialComponents) {
+        if (materialComponents.hasOwnProperty(prop) && AFRAME.GLTFModelPlus.components.hasOwnProperty(prop)) {
+          const { componentName, inflator } = AFRAME.GLTFModelPlus.components[prop];
+          await inflator(el, componentName, materialComponents[prop], materialComponents, indexToEntityMap);
```

Comment:
Kinda confused by what you mean here. The components are added to the entity that is the direct parent of the object3D, not the root of the avatar.

It also shouldn't have any issue with the existing material components we have since they are not registered in our gltf component mappings. This new method requires less component specific code in the gltf-model-plus component. Which we may want to use for things like lightmaps etc. going forward.